src/Arcus.Messaging.Pumps.Abstractions/Resiliency/DefaultMessagePumpCircuitBreaker.cs

```diff
@@ -44,25 +43,62 @@ public virtual async Task PauseMessageProcessingAsync(string jobId, Action<Messa
         {
             Guard.NotNullOrWhitespace(jobId, nameof(jobId));
 
-            _logger.LogTrace("Open circuit by pausing message processing for message pump '{JobId}'...", jobId);
-
             var options = new MessagePumpCircuitBreakerOptions();
             configureOptions?.Invoke(options);
 
             MessagePump messagePump = GetRegisteredMessagePump(jobId);
             if (messagePump.IsStarted)
             {
+                _logger.LogDebug("Open circuit by pausing message processing for message pump '{JobId}'...", jobId);
                 await messagePump.StopProcessingMessagesAsync(CancellationToken.None);
-                await WaitRecoveryTimeAsync(messagePump, options);
-                await TryProcessSingleMessageAsync(messagePump, options);
+
+                await Task.Factory.StartNew(async () =>
+                {
+                    await WaitRecoveryTimeAsync(messagePump, options);
+                    await TryProcessSingleMessageAsync(messagePump, options);
```

Comment:
We're trying to process a single message, but what should we do next ?
Shouldn't we continue trying if processing that single message fails ?
In other words, shouldn't these 2 lines (L57 & 58) be executed in a loop until the processing succeeded ?


/just thinking out loud.

Reply:
And once succeeded, can't the message-pump transition to the Closed state by itself ?

Reply:
I've tried this locally, and seem to have good results with it.  Additional benefit is that as an 'application programmer', I don't have to worry myself about resuming the pump / closing the circuitbreaker.

I changed this on my local version to this:

```
 if (messagePump.IsStarted)
 {
     _logger.LogInformation("Open circuit by pausing message processing for message pump '{JobId}'...", jobId);
     await messagePump.StopProcessingMessagesAsync(CancellationToken.None);

     await Task.Factory.StartNew(async () =>
     {
         _logger.LogInformation("Starting recovery-task");
         MessageProcessingResult result;
         do
         {
             await WaitRecoveryTimeAsync(messagePump, options);
             _logger.LogInformation("Done waiting in recovery");
             result = await TryProcessSingleMessageAsync(messagePump, options);
         } while (result.IsSuccessful == false);

         await ResumeMessageProcessingAsync(jobId); 
     }, TaskCreationOptions.LongRunning);
 }
 else
 {
     throw new InvalidOperationException("Cannot pause when not running");
 }
```